### DISCUSSION TITLE
Specify fields returned from UserInfo endpoint

### DISCUSSION THREAD

**Discussion by xperiandri** [AzureAD/microsoft-identity-web#1772] [Q&A]
Can I specify which fields about a user to return from UserInfo endpoint?

**Comment by jmprieur** [comment]
No, @xperiandri you cannot. There are only two, the home tenant Id and the home object ID of the user.
This should be an implementation detail. What are you trying to achieve?

**Reply by xperiandri** [reply]
I'm building an [OpenIddict-based](https://github.com/openiddict/openiddict-core) authentication server.
From each provider (Azure AD, Feide, Skolon) I try to gather the same info about the user:
* TenantId
* TenantUserId
* Display name
* First name
* Last name
* Email
* Groups that he/she belongs to (Teams groups in case of Teams)
* Affiliation (teacher/student), license in case of Teams to determine that

All that stuff I would like to put into principal claims on sign-in and then with a single code put into a user data models.

**Comment by jmprieur** [comment]
@xperiandri : you might want to have a look at the optional claims (that you would have to request in your app registration):
https://docs.microsoft.com/en-us/azure/active-directory/develop/active-directory-optional-claims
